testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergSparkDropTableCompatibility.java

```diff
@@ -83,7 +83,8 @@ public void testCleanupOnDropTable(Engine tableCreatorEngine, Engine tableDroppe
         List<String> dataFilePaths = getDataFilePaths(tableName);
 
         tableDropperEngine.queryExecutor().executeQuery("DROP TABLE " + tableName);
-        assertFileExistence(tableDirectory, false, format("The table directory %s should be removed after dropping the table", tableDirectory));
+        boolean expectExists = tableDropperEngine == SPARK; // Note: Spark's behavior is Catalog dependent
```

Comment:
What exactly is meant by the note:

> // Note: Spark's behavior is Catalog dependent

?

Reply:
Here we expect that "spark does not delete table data"
however this is only conditionally the case, see eg
https://github.com/apache/iceberg/blob/17c0ff7f698bced6f03711819073b26bb8bb3e09/spark/v3.2/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestRemoveOrphanFilesProcedure.java#L141-L148